src/components/TezosBlockView.tsx

```diff
@@ -0,0 +1,44 @@
+import * as React from 'react';
+import { getBlock } from '../Conseil';
+
+interface TezosBlockProps {
+    network: string;
+    id: string;
+}
+
+interface TezosBlockState {
+    data: object;
+}
+
+export class TezosBlockView extends React.Component<TezosBlockProps, TezosBlockState> {
+
+    constructor(props: TezosBlockProps) {
+        super(props);
+        this.state = {data: {}};
+        this.refreshData = this.refreshData.bind(this);
+        this.refreshData(props);
+    }
+
+    refreshData(props: TezosBlockProps) {
+        getBlock(this.props.network, this.props.id).
+        then(value => this.setState({data: JSON.parse(decodeURI(value))}));
+    }
+
+    componentWillReceiveProps(nextProps: TezosBlockProps) {
+        this.refreshData(nextProps);
+    }
+
+    public render() {
+        if (this.state.data === {}) {
```

Comment:
🤔 maybe react makes this possible, but in general in javascript `{} === {}` will always be false because it checks the object reference and not the object values. I wouldn't check object equality like this, but you could use some methods available in libraries like lodash and ramda. In this case, since you are just checking that the object is empty, you could just check `Object.keys(this.state.data).length === 0`

Reply:
Thanks, changed!